source/backend/routes/users.js

```diff
@@ -86,4 +86,27 @@ router.delete('/:savedRecipeId', async (req, res) => {
   }
 });
 
+/* POST /user/completedRecipes */
+router.post('/completedRecipes', async (req, res) => {
+  try {
+    const { userId } = req.userInfo;
+    const { recipeId } = req.recipeInfo;
+    if (parseInt(req.params.id, 10) !== userId) {
+      return res.status(401).json({ message: 'Forbidden, acceess denied' });
+    }
```

Comment:
Also, where is req.recipeInfo set? For the recipe id, it should come from the body of the request so something like req.body.recipeId. The req.userInfo is fine since will come from the token that is parsed in verifyUserToken().

Reply:
Thanks for clarifying, I was not aware that the verifyUserToken middleware already does a check and will throw a login error if userId is null. I removed the redundant check that I had here.